--- REST.JsonReflect.pas.org	Sat Feb 20 06:58:20 2021
+++ REST.JsonReflect.pas	Sat Sep 03 07:48:39 2022
@@ -1200,8 +1200,9 @@
   LFieldName := ConvertFieldNameToJson(Field);
   if (Current is TJSONPair) and (TJSONPair(Current).JsonString.Value = LFieldName) then
   begin
-    if TJSONPair(Current).JsonValue = nil then
-      raise EConversionError.Create(Format(SFieldValueMissing, [LFieldName]));
+	// LMR:
+    //if TJSONPair(Current).JsonValue = nil then
+    // raise EConversionError.Create(Format(SFieldValueMissing, [LFieldName]));
     FStack.Pop;
   end
   else
@@ -1735,7 +1736,8 @@
         FConverter.OnString(Value.AsString);
       TTypeKind.tkEnumeration:
         if ((fieldRTTI <> nil) and (string.CompareText('Boolean', fieldRTTI.FieldType.Name) = 0)) or
-          ((fieldRTTI = nil) and (string.CompareText('Boolean', Value.typeInfo.NameFld.ToString) = 0)) then
+		  // LMR:	
+          (string.CompareText('Boolean', Value.typeInfo.NameFld.ToString) = 0) then
           FConverter.OnBoolean(Value.AsBoolean)
         else
           FConverter.OnString(GetEnumName(Value.typeInfo, TValueData(Value).FAsSLong));
@@ -1801,7 +1803,9 @@
       TTypeKind.tkMethod:
         raise EConversionError.Create(Format(STypeNotSupported, ['tkMethod']));
       TTypeKind.tkVariant:
-        raise EConversionError.Create(Format(STypeNotSupported, ['tkVariant']));
+         // LMR:
+         FConverter.OnNull;
+         // raise EConversionError.Create(Format(STypeNotSupported, ['tkVariant']));
       TTypeKind.tkInterface:
         raise EConversionError.Create(Format(STypeNotSupported, ['tkInterface']));
       TTypeKind.tkClassRef:
@@ -3069,7 +3073,9 @@
                 else if jsonFieldVal is TJSONNull then
                   revAttr.ObjectsReverter(Data, LDlpFieldName, nil)
                 else
-                  raise EConversionError.Create(Format(SObjectExpectedForField, [FieldName, jsonFieldVal.ToString]));
+				  // LMR:
+                  revAttr.StringReverter(Data, LDlpFieldName, jsonFieldVal.ToString);
+                  //raise EConversionError.Create(Format(SObjectExpectedForField, [FieldName, jsonFieldVal.ToString]));
               end
           else
             raise EConversionError.Create(Format(SNoConversionForType,
@@ -3797,7 +3803,8 @@
 begin
   Assert(Data <> nil);
   // Under ARC there is always a refCount field which we DO NOT WANT serialized
-  if rttiField.Name = 'FRefCount' then
+  // LMR:
+  if (rttiField.Name = 'FRefCount') or (rttiField.Name.StartsWith('_'))  then
   begin
     Result := false;
   end
