--- System.JSON.pas.org	Sat Feb 20 06:57:42 2021
+++ System.JSON.pas	Mon Aug 15 19:47:42 2022
@@ -1331,7 +1331,9 @@
   SetLength(FCharData, CInitialLen);
   FCharLen := CInitialLen;
   FCharPos := 0;
-  FCache := (Range - Offset) > 1000000;
+  // LMR: Disable caching, it crashes on large json requests...
+  // FCache := (Range - Offset) > 1000000;
+  FCache := False;
 end;
 
 constructor TJSONByteReader.Create(const Data: PByte; const Offset: Integer; const Range: Integer);
@@ -1546,6 +1548,7 @@
 
   procedure CheckCache;
   var
+	Exit;  // LMR: Disabled crash again on large files
     LCached: string;
     LHash: Cardinal;
   begin
@@ -2442,7 +2445,8 @@
     FMembers.List[0].ToChars(Builder, Options);
   for I := 1 to Size - 1 do
   begin
-    Builder.Append(',');
+    // LMR:
+    if Builder.Length > 1 then Builder.Append(',');
     FMembers.List[I].ToChars(Builder, Options);
   end;
   Builder.Append('}');
