--- O:/Dev/Delphi/qBit4Delphi/API/JSON/22/System.JSON.pas.org	Wed Mar  9 03:38:36 2022
+++ O:/Dev/Delphi/qBit4Delphi/API/JSON/22/System.JSON.pas	Mon Aug 15 19:14:54 2022
@@ -1387 +1387,3 @@ begin
-  FCache := (Range - Offset) > 1000000;
+  // LMR: Disable caching, it crashes on large json requests...
+  // FCache := (Range - Offset) > 1000000;
+  FCache := False;
@@ -1595 +1597 @@ procedure TJSONByteReader.FlushString(var ADest: s
-      Result := (Result xor Ord(P^)) * 16777619;
+      Result := (Result xor Ord(P^)) * 16777619; // <<  * 16777619 = Integer Overflow : this has not been tested !!!
@@ -1604,0 +1607 @@ procedure TJSONByteReader.FlushString(var ADest: s
+    Exit;  // LMR: Disabled crash again on large files
@@ -2885,2 +2888,6 @@ begin
-    Builder.Append(',');
-    FMembers.List[I].ToChars(Builder, Options);
+    if FMembers.List[I].JsonValue<> nil then
+    begin
+      // LMR:
+      if Builder.Length > 1 then Builder.Append(',');
+      FMembers.List[I].ToChars(Builder, Options);
+    end;
