--- System.JSON.pas	Tue Aug 31 07:00:24 2021
+++ System.JSON.pas	Thu Oct 28 06:25:42 2021
@@ -1379,7 +1379,9 @@
   SetLength(FCharData, CInitialLen);
   FCharLen := CInitialLen;
   FCharPos := 0;
-  FCache := (Range - Offset) > 1000000;
+  // LMR: Disable caching, it crashes on large json requests...
+  // FCache := (Range - Offset) > 1000000;
+  FCache := False;
 end;
 
 constructor TJSONByteReader.Create(const Data: PByte; const Offset: Integer; const Range: Integer);
@@ -1587,7 +1589,7 @@
     PEnd := P + Result;
     while P < PEnd do
     begin
-      Result := (Result xor Ord(P^)) * 16777619;
+      Result := (Result xor Ord(P^)) * 16777619; // <<  * 16777619 = Integer Overflow : this has not been tested !!!
       Inc(P);
     end;
   end;
@@ -2877,8 +2879,12 @@
     FMembers.List[0].ToChars(Builder, Options);
   for I := 1 to Size - 1 do
   begin
-    Builder.Append(',');
-    FMembers.List[I].ToChars(Builder, Options);
+    if FMembers.List[I].JsonValue<> nil then
+    begin
+      // LMR:
+      if Builder.Length > 1 then Builder.Append(',');
+      FMembers.List[I].ToChars(Builder, Options);
+    end;
   end;
   Builder.Append('}');
 end;
