--- REST.Json.pas	Sat Feb 20 05:58:20 2021
+++ REST.Json.pas	Fri Dec 03 17:13:46 2021
@@ -241,6 +241,14 @@
   LItem: TObject;
   i: Integer;
 begin
+  // LMR:
+  for i := AJsonObject.Count - 1 downto 0 do
+  begin
+    LPair := TJSONPair(AJsonObject.Pairs[i]);
+    if LPair.JsonValue is TJSONNull then
+       AJsonObject.RemovePair(LPair.JsonString.Value).DisposeOf
+  end;
+  //
   if not Assigned(AJsonObject) or (AOptions * [joIgnoreEmptyStrings, joIgnoreEmptyArrays] = []) then
     Exit;
 
@@ -258,9 +266,10 @@
           if LItem is TJSONObject then
             ProcessOptions(TJSONObject(LItem), AOptions)
     end
-    else
+    else 
       if (joIgnoreEmptyStrings in AOptions) and (LPair.JsonValue.Value = '') then
         AJsonObject.RemovePair(LPair.JsonString.Value).DisposeOf;
+
   end;
 end;
 
